.github/workflows/test-pki.yml

```diff
@@ -0,0 +1,22 @@
+name: Test PKI
+on:
+  push:
+  pull_request:
+
+jobs:
+  test-pki:
+    name: test-pki
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v4.1.0
+        with:
+          go-version: '~1.19'
+      - name: Setup test deps
+        uses: ConorMacBride/install-package@v1.1.0
```

Comment:
## Concerns about the Proposed GitHub Action

**Question:** Why are we considering using this particular action?

In our standard workflow configuration, we employ the following GitHub Actions:

\```yaml
steps:
  - name: Checkout code
    uses: actions/checkout@v2

  - name: Set up Go
    uses: actions/setup-go@v2
    with:
      go-version: '~1.19'
\```

I acknowledge the possibility of other GitHub Actions replicating the functionality of `actions/setup-go`, but I have certain reservations about adopting [ConorMacBride/install-package](https://github.com/ConorMacBride/install-package):

1. **Trust and Ownership:** The action is maintained by an individual without affiliation to a recognized organization or entity. This raises concerns about trustworthiness and accountability.
 
2. **Community Adoption:** A reliable indicator of a tool's dependability is its widespread use and endorsement by the developer community. This action does not seem to have garnered such widespread adoption.

3. **Sustainability Concerns:** When tools are maintained by individuals, there's an inherent risk of discontinuation. If the maintainer chooses to no longer update or support the action, we could be left with an unsupported tool in our workflow.

Given these points, I'd suggest sticking with the widely accepted `actions/setup-go` action unless there are overriding advantages in the alternative that we shouldn't overlook. Please, let's not use things that are not really adopted in the community.

Reply:
I'm not sure how `setup-go` can install apt packages.
Can you propose a solution?

I could surely run the plain script which calls `apt install $X`.
But, a ready to go action is better at maintaining the cache and resolving all specifics.
As for the _discontinuation_, I've seen two dozens of actions like this; not a concern for me.

That said, if the team prefers a script calling `apt install $X`, I'm fine with that.

Reply:
> I'm not sure how setup-go can install apt packages.

So, if the goal is only install the packages into the Linux env I would prefer to see that done as it usually is (as the following example). I think my main reasons are:
- That makes clear what is required and what is doing so that I can either do manually if I need to do so and to test something (make easier we maintain this one)
- Do not rely on a GitHub action done by an individual that nobody uses or knows how to use.


Example:
```
name: My Workflow

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install dependencies
      run: |
        sudo apt-get update
        sudo apt-get install -y <your-package-name>
```

Reply:
https://github.com/foundriesio/fioctl/pull/278/commits/98a7d6bb2ea9a2c7b271f2f553e2fe7816c2a6cc